src/hotspot/share/code/SCCache.cpp

```diff
@@ -380,59 +381,29 @@ bool SCCache::allow_const_field(ciConstant& value) {
         ;
 }
 
-bool SCCache::open_cache(const char* cache_path) {
-  if (LoadCachedCode) {
-    log_info(scc)("Trying to load Startup Code Cache '%s'", cache_path);
-    struct stat st;
-    if (os::stat(cache_path, &st) != 0) {
-      log_warning(scc, init)("Specified Startup Code Cache file not found '%s'", cache_path);
-      return false;
-    } else if ((st.st_mode & S_IFMT) != S_IFREG) {
-      log_warning(scc, init)("Specified Startup Code Cache is not file '%s'", cache_path);
-      return false;
-    }
-    int fd = os::open(cache_path, O_RDONLY | O_BINARY, 0);
-    if (fd < 0) {
-      if (errno == ENOENT) {
-        log_warning(scc, init)("Specified Startup Code Cache file not found '%s'", cache_path);
-      } else {
-        log_warning(scc, init)("Failed to open Startup Code Cache file '%s': (%s)", cache_path, os::strerror(errno));
-      }
-      return false;
-    } else {
-      log_info(scc, init)("Opened for read Startup Code Cache '%s'", cache_path);
-    }
-    SCCache* cache = new SCCache(cache_path, fd, (uint)st.st_size);
-    bool failed = cache->failed();
-    if (::close(fd) < 0) {
-      log_warning(scc)("Failed to close for read Startup Code Cache file '%s'", cache_path);
-      failed = true;
-    }
-    if (failed) {
-      delete cache;
-      _cache = nullptr;
-      return false;
-    }
-    _cache = cache;
-  }
-  if (_cache == nullptr && StoreCachedCode) {
-    SCCache* cache = new SCCache(cache_path, -1 /* fd */, 0 /* size */);
-    if (cache->failed()) {
-      delete cache;
-      _cache = nullptr;
-      return false;
-    }
-    _cache = cache;
+
+bool SCCache::open_cache() {
+  SCCache* cache = new SCCache();
+  if (cache->failed()) {
+    delete cache;
+    _cache = nullptr;
+    return false;
   }
+  _cache = cache;
   return true;
 }
 
 class CachedCodeDirectory : public CachedCodeDirectoryInternal {
 public:
-  int _some_number;
-  InstanceKlass* _some_klass;
-  size_t _my_data_length;
-  void* _my_data;
+  uint _aot_code_size;
+  char* _aot_code_data;
+
+  void set_aot_code_data(uint size, char* aot_data) {
+    _aot_code_size = size;
+    CDSAccess::set_pointer(&_aot_code_data, aot_data);
+  }
+
+  static CachedCodeDirectory* create();
 };
 
 // Skeleton code for including cached code in CDS:
```

Comment:
The comment "Skeleton code for including cached code in CDS" should be changed to something like "Storing AOT code in CDS"